### ISSUE TITLE
Request Handling Documentation Gone?

### ISSUE THREAD

**Issue by xanderdunn** [agrafix/Spock#118]
In an [old version of spock](https://hackage.haskell.org/package/Spock-0.6.0.0/docs/Web-Spock.html) on Hackage I can find a Request Handling section that includes information on `param`, `params`, `jsonBody`, etc.  There isn't information for any of these functions in the Hackage documentation for the most recent version of Spock.  Spock still exports these functions, why is the documentation missing?

**Comment by brynedwards** [comment]
Hi, those functions were moved to a separate package, `Spock-core`. You can find the Request Handling section [here](https://hackage.haskell.org/package/Spock-core-0.12.0.0/docs/Web-Spock-Action.html#g:2). I know this could be more obvious, I'm trying to help with improving docs at the moment.

**Comment by agrafix** [comment]
Yeah, all these functions are now "hidden" behind `module Web.Spock.Action`. Maybe we should reexport all the individual functions for good documentation, but then when making changes to `Web.Spock.Action` one has to be real careful to update all the reexports, so not a big fan...

**Comment by brynedwards** [comment]
I've been looking into doing this "the right way", which IMO would be showing the entire contents of `Web.Spock.Action` in `Web.Spock` without having to manually reexport everything. This is described in the [Haddock docs](https://www.haskell.org/haddock/doc/html/ch03s04.html#idm140354810861024) but doesn't seem to work, maybe due to this issue haskell/haddock/issues/563? Unless/until there is a way to make this work, I suggest either:

* Reexporting everything in `Web.Spock.Action` individually.
* Including a sentence or two below the actions heading explaining that the documentation for actions is in `Web.Spock.Action`

**Comment by agrafix** [comment]
Alternatively we could introduce a dummy identifier and then use `hiding` when importing/reexporting it. That could work?

**Comment by brynedwards** [comment]
I think that _should_ work, but I tried using `hiding` myself and couldn't get it to work, I'm assuming it's another Haddock issue.

**Comment by agrafix** [comment]
Hm... Then maybe just highlighting a link to `Spock-core` is the way to go for now.

**Comment by brynedwards** [comment]
Looking at this again, since `Web.Spock` and `Web.Spock.Core` are nearly identical, I think a nice layout would be to have `Web.Spock` reexport everything from `Web.Spock.Core` and have `module Web.Spock.Core` and `module Web.Spock.Action` at the top of `Web.Spock` followed by `Web.Spock` exports, like [this module](https://hackage.haskell.org/package/persistent-2.6.1/docs/Database-Persist.html) does.
